lib/ansible/plugins/filter/to_nice_json.yml

```diff
@@ -0,0 +1,64 @@
+DOCUMENTATION:
+  name: to_nice_json
+  author: core team
+  version_added: 'historical'
+  short_description: Convert variable to 'nicely formatted' JSON string
+  description:
+    - Converts an Ansible variable into a 'nicely formatted' JSON string representation. This filter functions as a wrapper to the Python ``json.dumps()`` function.
+    - Ansible internally auto-converts JSON strings into variable structures so this plugin is used to force it into a JSON string.
+  options:
+    _input:
+      description: A variable or expression that returns a data structure
+      type: raw
+      required: true
+      format: positional
+    vault_to_text:
+      description: Toggle to either unvault a vault or create the JSON version of a vaulted object
+      type: bool
+      default: True
+      format: named
+      version_added: '2.9'
+    preprocess_unsafe:
+      description: Toggle to represent unsafe values directly in JSON or create a unsafe object in JSON
+      type: bool
+      default: True
+      format: named
+      version_added: '2.9'
+    allow_nan:
+      description: When off, strict adherence to float value limits of the JSON spec, so ``nan``, ``-inf`` and ``-inf`` values will produce errors
+                   If on, JavaScript equivalents will be used (``NaN``, ``Infinity``, ``-Infinity``)
+      default: True
+      type: bool
+      format: named
+      provider: python's json.dumps()
```

Comment:
What should the `provider` field be in general? If this shouldn't be something that's used by other plugins, this should be moved to `description` instead.

Reply:
i had `provider: ansbile` for the others, but then realized that should be the default. Also i left 'per option' as an easy identifier, jic we also have more than one provider per plugin (but have not found a case yet)